featuretools/computational_backends/pandas_backend.py

```diff
@@ -341,9 +358,11 @@ def _calculate_direct_features(self, features, entity_frames):
             # Sometimes entityset._add_multigenerational_links adds link variables
             # that would ordinarily get calculated as direct features,
             # so we make sure not to attempt to calculate again
-            if f.get_name() in child_df.columns:
-                continue
-            col_map[f.base_features[0].get_name()] = f.get_name()
+            base_names = f.base_features[0].get_feature_names()
```

Comment:
you can do just `f.base_feature.get_feature_names()`

Reply:
Changing this causes an error related to this property direct features have that `DirectFeature.base_feature` points to the desired feature while `DirectFeatures.base_features[0]` points to the next feature on the path between the base feature and the child.  So `sessions.customers.loves_ice_cream` has `<Feature: loves_ice_cream>` as the `base_feature` but `customers.loves_ice_cream` for `base_features[0]`.

I saw a TODO above the logic for how `DirectFeature` does this, maybe I should add an explanation?